qiskit/providers/ibmq/job/ibmqjob.py

```diff
@@ -344,8 +344,13 @@ def error_message(self):
 
         if not self._api_error_msg:
             job_response = self._get_job()
-            results = job_response['qObjectResult']['results']
-            self._api_error_msg = build_error_report(results)
+            if 'qObjectResult' in job_response:
+                results = job_response['qObjectResult']['results']
+                self._api_error_msg = build_error_report(results)
+            elif 'qasms' in job_response:
+                qasm_statuses = [qasm['status'] for qasm in job_response['qasms']]
+                self._api_error_msg = 'Job resulted in the following QASM status(es): ' \
+                                      '{}.'.format(', '.join(qasm_statuses))
```

Comment:
Can you reintroduce the third `else: ` block? It might be hard to reach, but it would guard against odd responses that do not contain `qObjectResult` or `qasms` in the content. For extra safety:

```python
else:
                self._api_error_msg = job_response.get('status', 'An unknown error occurred.')
```

(untested line!) The intent of the line is to make use of the `status` field if present (as the API might return useful information that we can present to the user other than the default test, and the `else` makes sure that we write something into `self._api_error_msg` one way or the other.